components/core/CMakeLists.txt

```diff
@@ -130,7 +130,13 @@ else()
 endif()
 
 # Find and setup msgpack
-find_package(msgpackc-cxx 5.0.0 REQUIRED)
+if(APPLE)
+    execute_process(COMMAND brew --prefix msgpack-cxx OUTPUT_VARIABLE PREFIX_MSGPACK_CXX)
```

Comment:
`msgpack-cxx` installed through brew does not come with the pkg-config file, so it cannot be automatically detected.

Reply:
Based on our offline discussion, do you think this is necessary if we add msgpack to our macOS dependencies?

Reply:
Based on our offline discussion, do you think this is necessary if we add msgpack to our macOS dependencies?

Reply:
I've done a bit of searching and it seems like a name mix-up on the `msgpack` side ([msgpack#974](https://github.com/msgpack/msgpack-c/issues/974)). They are aware of the issue and they are actively fixing it (see [msgpack-c#1043](https://github.com/msgpack/msgpack-c/issues/1043), only closed last week). They released the fix in a higher version, i.e., 6.0.0. But unfortunately, [msgpack-cxx v6.0.0](https://github.com/Homebrew/homebrew-core/blob/ceaf4b973958cf7ee9ce1197b8700357c6d80dc4/Formula/msgpack-cxx.rb) still does not have a `.pc` file.

Therefore, I would recommend `brew --prefix` approach for now instead of the error-prone co-existence of both `c` and `cxx` library.